spec/ucb/tools/batch_page_spec.rb

```diff
@@ -0,0 +1,174 @@
+      end
+
+      it 'should not be able to alter the data type contexts' do
+        @batch_page.click_invocable @test_0[CoreInvocablesData::INVOCABLE_NAME.name]
+        @batch_page.uncollapse_panel_if_collapsed CoreInvocablesData::INVOCABLE_RUNS_ON_PANEL.label
+        expect(@batch_page.enabled? @batch_page.invocable_no_ctx_locator).to be false
+        expect(@batch_page.enabled? @batch_page.invocable_single_ctx_locator).to be false
+        expect(@batch_page.enabled? @batch_page.invocable_group_ctx_locator).to be false
+        expect(@batch_page.enabled? @batch_page.invocable_list_ctx_locator).to be false
+        expect(@batch_page.enabled? @batch_page.revert_button).to be false
+        expect(@batch_page.enabled? @batch_page.run_button).to be true
+      end
+
+      it 'should not be able to alter the record types' do
+        @batch_page.click_invocable @test_0[CoreInvocablesData::INVOCABLE_NAME.name]
+        @batch_page.uncollapse_panel_if_collapsed CoreInvocablesData::INVOCABLE_RUNS_ON_PANEL.label
+        expect(@batch_page.enabled? @batch_page.invocable_doctypes_locator).to be false
+        expect(@batch_page.enabled? @batch_page.revert_button).to be false
+        expect(@batch_page.enabled? @batch_page.run_button).to be true
+      end
+
+      it 'should not be able to alter whether the data update creates a new focus' do
+        @batch_page.click_invocable @test_0[CoreInvocablesData::INVOCABLE_NAME.name]
+        @batch_page.uncollapse_panel_if_collapsed CoreInvocablesData::INVOCABLE_RUNS_ON_PANEL.label
+        expect(@batch_page.enabled? @batch_page.invocable_batch_new_focus_locator).to be false
+        expect(@batch_page.enabled? @batch_page.revert_button).to be false
+        expect(@batch_page.enabled? @batch_page.run_button).to be true
+      end
+    end
+
+    context 'Data updates UI interactions' do
+
+      it 'should be able to filter based on a search' do
+        @tools_page.fill_filter_bar(@test_0[CoreInvocablesData::INVOCABLE_NAME.name])
+        expect(@search_results_page.row_exists? @test_0[CoreInvocablesData::INVOCABLE_NAME.name]).to be true
+        expect(@search_results_page.row_exists? @test_1[CoreInvocablesData::INVOCABLE_NAME.name]).to be false
+      end
+
+      it 'should show all results when the clear button is clicked' do
+        @tools_page.fill_filter_bar(@test_0[CoreInvocablesData::INVOCABLE_NAME.name])
+        expect(@search_results_page.row_exists? @test_0[CoreInvocablesData::INVOCABLE_NAME.name]).to be true
+        expect(@search_results_page.row_exists? @test_1[CoreInvocablesData::INVOCABLE_NAME.name]).to be false
+
+        @tools_page.click_clear_button
+        expect(@search_results_page.row_exists? @test_0[CoreInvocablesData::INVOCABLE_NAME.name]).to be true
+        expect(@search_results_page.row_exists? @test_1[CoreInvocablesData::INVOCABLE_NAME.name]).to be false
+      end
+
+      it 'should be able to bring up a modal and dismiss it in various ways' do
```

Comment:
@cesarvh - I would suggest breaking this large modal-dismissing example into three parts.  If one dismissal fails, then the other(s) will never be tested.